microsoft/surface/firmware/surface-go/ath10k/default.nix

```diff
@@ -0,0 +1,22 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.hardware.microsoft-surface.firmware.surface-go-ath10k;
+in
+{
+  options = {
+    hardware.microsoft-surface.firmware.surface-go-ath10k = {
+      replace = lib.mkEnableOption ''Use the "board.bin" firmware for ath10k-based WiFi on Surface Go.'';
```

Comment:
I renamed this from `enable` to `replace` for a clearer option-name.
This will hopefully avoid name-clashes with future alternative fixes for this problem.

e.g. there's an experimental kernel patch being discussed, which will hopefully support a way to non-destructively use the Surface Go firmware blob:
- https://github.com/linux-surface/kernel/commit/22ef83836c4aa89e9eb98de9b47ed24b6c2a1d45

Reply:
ok.